### PR TITLE
SCL-2584 Suggest to import scala.collection.JavaConversions._

### PR DESCRIPTION
**Pull request by MotiejusJuodelis**
SCL-2584 Suggest to import scala.collection.JavaConversions._ where necessary.
Does it makes sense to have JavaConversions though, since it's deprecated?

### CODE REVIEW

**Comment by unkarjedy** [comment]
relates to :
https://youtrack.jetbrains.com/issue/SCL-2584
https://youtrack.jetbrains.com/issue/SCL-15352

File: scala/scala-impl/src/org/jetbrains/plugins/scala/codeInsight/intention/types/ImplicitConvertJavaToScalaCollectionIntention.scala [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "java.util.Map"
+  )
+
+  override val alreadyConvertedPrefixes: Set[String] = Set("scala.collections")
+
+  override def getText: String = ScalaBundle.message("implicit.java.to.scala.conversions.hint")
+
+  override def getFamilyName: String = ImplicitConvertScalaToJavaCollectionIntention.getFamilyName
+
+  override val importPath: String = "scala.collection.JavaConversions._"
```
Comment (unkarjedy) [inline scala/scala-impl/src/org/jetbrains/plugins/scala/codeInsight/intention/types/ImplicitConvertJavaToScalaCollectionIntention.scala]: `scala.collection.JavaConversion._` is deprecated in 2.13.0
(though the problem applies to ConvertJavaToScalaCollectionIntention as well)

**Review by unkarjedy — COMMENTED** [review]
Thanks for the pull request

**Comment by jozic** [comment]
Aren't JavaConversions deprecated in 2.10-2.12 and removed in 2.13?

**Comment by julienrf** [comment]
Indeed, you should suggest importing `scala.jdk.CollectionConverters._`

https://www.scala-lang.org/api/2.13.1/scala/jdk/CollectionConverters$.html

**Comment by jozic** [comment]
> Indeed, you should suggest importing `scala.jdk.CollectionConverters._`
> 
> https://www.scala-lang.org/api/2.13.1/scala/jdk/CollectionConverters$.html

the tricky part is that for 2.11-2.12 they come in a separate lib (and not even available for 2.10)
if intellij can figure out if those are in classpath and suggest using those over current suggestion of JavaConverters, that would be neat

**Comment by niktrop** [comment]
@MotiejusJuodelis I think we should not encourage implicit conversions between java and scala collections and suggest only explicit `asJava` and `asScala` method calls. 

I've created a ticket about changes in 2.13:
https://youtrack.jetbrains.com/issue/SCL-16730

Do you want to implement it? You may use an extension method `ProjectPsiElementExt#scalaLanguageLevel` to choose correct import.

**Comment by MotiejusJuodelis** [comment]
> @MotiejusJuodelis I think we should not encourage implicit conversions between java and scala collections and suggest only explicit `asJava` and `asScala` method calls.
> 
> I've created a ticket about changes in 2.13:
> https://youtrack.jetbrains.com/issue/SCL-16730
> 
> Do you want to implement it? You may use an extension method `ProjectPsiElementExt#scalaLanguageLevel` to choose correct import.

@niktrop yes, agree. I'm planning to come back to this and implement https://youtrack.jetbrains.com/issue/SCL-16730
Thanks

**Comment by SrTobi** [comment]
@MotiejusJuodelis hey cool, write to me you need help

**Pull request closed without merging**